PMSCore/ReportManager.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using PMSCore;
+
+namespace PMSCore
+{
+    /// <summary>
+    /// This class handles the creation and management of reports.
+    /// </summary>
+    public sealed class ReportManager
+    {
+        // The single instance of ReportManager
+        private static ReportManager _instance;
+
+        // Logger instance
+        private readonly ILogger _logger;
+
+        /// <summary>
+        /// Private constructor to prevent external instantiation.
+        /// </summary>
+        private ReportManager()
+        {
+            // Initialize the logger here
+            _logger = LoggerFactory.CreateLogger<ReportManager>();
+            _logger.Log("ReportManager instance created.");
+        }
+
+        /// <summary>
+        /// Provides access to the single instance of ReportManager.
+        /// </summary>
+        /// <returns>The single instance of ReportManager.</returns>
+        public static ReportManager GetInstance()
+        {
+            if (_instance == null)
+            {
+                _instance = new ReportManager();
+            }
+
+            return _instance;
+        }
+
+        /// <summary>
+        /// Resets the instance of ReportManager for reinitialization.
+        /// </summary>
+        public static void ResetInstance()
+        {
+            _instance = null;
+        }
+
+        /// <summary>
+        /// Creates a report from the provided project.
+        /// </summary>
+        /// <param name="project">The project to create the report from.</param>
+        /// <returns>A string containing the project's report.</returns>
+        /// <exception cref="ArgumentNullException">Thrown if the project is null.</exception>
+        public string GenerateProjectReport(Project project)
+        {
+            if (project == null)
+            {
+                throw new ArgumentNullException(nameof(project), "Project cannot be null.");
+            }
+
+            _logger.Log("Generating project report.");
+            return project.GenerateReport();
+        }
+
+        /// <summary>
+        /// Creates a report from the provided task.
+        /// </summary>
+        /// <param name="task">The task to create the report from.</param>
+        /// <returns>A string containing the task's report.</returns>
+        /// <exception cref="ArgumentNullException">Thrown if the task is null.</exception>
+        public string GenerateTaskReport(Task task)
+        {
+            if (task == null)
+            {
+                throw new ArgumentNullException(nameof(task), "Task cannot be null.");
```

Comment:
Any reason to throw here?

Reply:
@DistroRickDev , I was thinking it would be okay to make sure the method gets valid input and to avoid possible errors later. With ArgumentNullException it's hopefully easier to fix the code later if needed because we know what went wrong. Let me know what you think.